api/types/swarm/secret.go

```diff
@@ -0,0 +1,30 @@
+package swarm
+
+import "os"
+
+// Secret represents a secret.
+type Secret struct {
+	ID string
+	Meta
+	Spec       SecretSpec
+	Digest     string
+	SecretSize int64
+}
+
+type SecretSpec struct {
+	Annotations
+	Data []byte
+}
+
+type SecretReferenceFileTarget struct {
+	Name string
+	UID  string
+	GID  string
+	Mode os.FileMode
+}
+
+type SecretReference struct {
+	SecretID   string
+	SecretName string
+	Target     SecretReferenceFileTarget
```

Comment:
This may need to be a pointer type. In the future, there may be EnvTarget, SystemTarget, etc.